src/main/java/classify/data/DataReader.java

```diff
@@ -106,15 +106,15 @@ public static void restoreStudentList(String studentFileInput, ArrayList <Studen
                     DataUI.noSubjectMessage();
                 } else {
 
-                for (int i = 0; i < numberOfSubjects ; i++){
-                    String [] subjectDetailedInfo = allSubjects[i].split(SUBJECT_INFO_REGEX);
-                    //Subject Name
-                    String subjectName = subjectDetailedInfo[ZERO].trim();
-                    InputParsing.checkForSpecialCharacters(subjectName);
-                    //Subject Grade
-                    double subjectGrades = Double.parseDouble(subjectDetailedInfo[ONE].trim());
-                    //Classes attended for this subject
-                    int subjectClassesAttended = Integer.parseInt(subjectDetailedInfo[TWO].trim());
+                    for (int i = 0; i < numberOfSubjects ; i++){
+                        String [] subjectDetailedInfo = allSubjects[i].split(SUBJECT_INFO_REGEX);
```

Comment:
Simple and quick fix!
![image](https://github.com/AY2324S2-CS2113-T13-3/tp/assets/110646458/50fa0786-6af8-4bf1-8063-15a828a60a91)